### PR TITLE
RTT example - allow empty lines too

### PR DESCRIPTION
**Pull request by fanoush**
It is typical to just press enter to see if console input is working,
I was very confused that nothing happens so connected to my device via UART to produce some output to RTT,
and only after I found it working I had an idea to actually type some text before pressing enter.

BTW  I could  not test it but hopefully it works, I cloned repo and set up github pages but when trying at https://fanoush.github.io/dapjs/examples/rtt/web.html the page gets 404 on https://fanoush.github.io/dapjs/dist/dap.umd.js
Not sure how I should put stuff to "dist"

EDIT: now I tested it by committing directly to my gh-pages branch and it seems to work

### CODE REVIEW

**Comment by fanoush** [comment]
And BTW I have also added some other stuff to the RTT example, not sure if you would be interested in some of them? you  may check https://github.com/ARMmbed/dapjs/compare/gh-pages...fanoush:dapjs:gh-pages
- faster ram searching
- direct character mode
- adding VID for [Raspberry Pi Pico debugprobe](https://github.com/raspberrypi/debugprobe) - which is cheap and capable CMSIS-DAP v2 debugger anyone can make from spare RP2040/Pico board
- avoid CPU halt/reset this should not be needed for RTT at all and it interferes with device that is already running something

I am testing it with Espruino build with interactive console over RTT and it seems to work pretty well. Character mode allows tab completion, ctrl+c break etc

The IRQ button change is not that useful but in case of Espruino it wakes device from sleep when activating the console after first keypress, unfortunately Segger RTT is designed to be polled from both sides instead if triggering interrupt when data comes so there is no way to wake the device otherwise. After first input espruino imlementation starts polling timer so then it works until RTT console is deactivated.

File: examples/rtt/web.html [hunk 1]
```diff
@@ -120,12 +120,13 @@
             const printable = !ev.altKey && !ev.ctrlKey && !ev.metaKey;
 
             if (ev.keyCode === 13) { // process newline
-                if (line_input.length) {
+                // allow empty lines too
+                // if (line_input.length) {
```
Comment (thegecko) [inline examples/rtt/web.html:124]: Can you remove the commented out lines, please?

File: examples/rtt/web.html [hunk 2]
```diff
@@ -120,12 +120,13 @@
...
                     term.write('\r\n');
-                    cmd = line_input;
+                    cmd = line_input + "\r\n";
                     console.log('sending: \"' + line_input + '\"');
                     line_input = "";
-                }
+                //}
```
Comment (thegecko) [inline examples/rtt/web.html:129]: ditto

**Review by thegecko — CHANGES_REQUESTED** [review]

File: examples/rtt/web.html (hunk 1 above)
Comment (fanoush) [reply examples/rtt/web.html]: done

**Comment by thegecko** [comment]
Thanks for your PR and sorry for the delay...

**Pull request merged**